pod.go

```diff
@@ -389,6 +389,11 @@ func (p *Pod) Annotations(key string) (string, error) {
 	return value, nil
 }
 
+// SetOrAddAnnotation sets or adds an annotation
+func (p *Pod) SetOrAddAnnotation(key, value string) {
+	p.config.Annotations[key] = value
```

Comment:
@sboeuf - agreed on both points.

@devimc - I think it makes sense to have an API to set annotations (much safer than simply altering the state of the objects directly), but do you have a particular use-case in mind by adding this (if so, that might help us all to hone in on the best design)?